src/main/java/com/netflix/conductor/kafka/eventqueue/KafkaObservableQueue.java

```diff
@@ -0,0 +1,408 @@
+                if (this.filteringHeader != null && !this.filteringValue.equals(headers.get(this.filteringHeader))
+                        || this.filterByPartitionKey && !this.filteringValue.equals(record.key())) {
+                    addAsyncCommitOffset(record.topic(), record.partition(), record.offset());
+                    return;
+                }
+                String headersJson = null;
+                try {
+                    headersJson = objectMapper.writeValueAsString(headers);
+                } catch (JsonProcessingException e) {
+                    logger.error("message headers build error.", e);
+                }
+                logger.debug("Consumer Record: key: {}, value: {}, partition: {}, offset: {}",
+                        record.key(), record.value(), record.partition(), record.offset());
+                String id = record.key() + ":" + record.topic() + ":" + record.partition() + ":" + record.offset();
+                String messagePayload = String.format("{\"eventData\":%s,\"eventHeaders\":%s}",
+                        record.value(), headersJson);
+                Message message = new Message(id, messagePayload, "");
+                messages.add(message);
+            });
+            if (messages.isEmpty()) {
+                commitAsyncOffsets();
+            }
+            Monitors.recordEventQueueMessagesProcessed(QUEUE_TYPE, this.queueURI, messages.size());
+        } catch (KafkaException e) {
+            logger.error("kafka consumer message polling failed.", e);
+            Monitors.recordObservableQMessageReceivedErrors(QUEUE_TYPE);
+        }
+        return messages;
+    }
+
+    void addAsyncCommitOffset(String topic, int partition, long offset) {
+        TopicPartition topicPartition = new TopicPartition(topic, partition);
+        commitOffsetMap.get().compute(topicPartition, (k, v) -> v == null ? offset : Math.max(v, offset));
+    }
+
+    void commitAsyncOffsets() {
+        Map<TopicPartition, OffsetAndMetadata> currentOffsets;
+        for (Map.Entry<TopicPartition, Long> offsetEntry : commitOffsetMap.get().entrySet()) {
+            currentOffsets = new HashMap<>();
+            currentOffsets.put(offsetEntry.getKey(),
+                    new OffsetAndMetadata(offsetEntry.getValue() + 1, "no metadata"));
+            try {
+                consumer.commitSync(currentOffsets);
+            } catch (KafkaException ke) {
+                logger.error("kafka consumer filtered messages commit failed.", ke);
+            }
+        }
+        commitOffsetMap.get().clear();
+    }
+
+    /**
+     * Publish the messages to the given topic.
+     *
+     * @param messages
+     */
+    void publishMessages(List<Message> messages) {
+
+        if (messages == null || messages.isEmpty()) {
+            return;
+        }
+        for (Message message : messages) {
+            final ProducerRecord<String, String> record = new ProducerRecord<>(queueName, message.getId(),
```

Comment:
I thought that queueName name can be set right on conductor task configuration, so it should be taken from message paramerters?

Reply:
Its value is obtained from sink parameter of EVENT task

Reply:
Can it be taken from configuration?

Reply:
No, we have only message list